### PR TITLE
Writing your first unit test with pytest

### PR DESCRIPTION
**Pull request by lkingsford**

### CODE REVIEW

File: articles/2020-lkingsford_writing-your-first-unit-test-with-pytest.md [hunk 1]
```diff
@@ -25,11 +25,51 @@ in Python's standard library - as it provides some useful out-of-the-box
...
 for them.
 
+#### Creating the environment
+
+- If you haven't already, create a folder for your project
+- Go into your terminal, and navigate to the folder
+- Create a virtual environment in a folder called `env` by running
+  `python3 -m venv env` in your terminal
+- Activate the virtual environment by running ` in your
+  terminal
```
Comment (nathanruiz) [inline articles/2020-lkingsford_writing-your-first-unit-test-with-pytest.md:46]: Do you want this point here given the next section?

File: articles/2020-lkingsford_writing-your-first-unit-test-with-pytest.md [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+After all - if you want to test your string manipulation function with the
+entire [Big List of Naughty Strings](https://github.com/minimaxir/big-list-of-naughty-strings)
+you probably don't want to make a few hundred copies of the same function.
+
+Lots of testing frameworks - including Pytest - support a feature called
+'parametrization'. This means that can get data from an iterator and run a
+test with that data
+
+> Yes, you can use generators - as your can use any iterators. It's outside the
```
Comment (nathanruiz) [inline articles/2020-lkingsford_writing-your-first-unit-test-with-pytest.md:310]:
```suggestion
> Yes, you can use generators - as you can use any iterators. It's outside the
```

File: articles/2020-lkingsford_writing-your-first-unit-test-with-pytest.md [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+So - here is our new version of `tests/test_user.py` that uses parametrization
+to test all combinations of input.
+
+```python
+import pytest
+import models.user
+
+
+TEST_DISPLAY_NAME_DATA = [
+    ['username', None,None, 'username'],
```
Comment (nathanruiz) [inline articles/2020-lkingsford_writing-your-first-unit-test-with-pytest.md:324]: Should you have tuples for these values?
```suggestion
    ('username', None,None, 'username'),
```

Comment (lkingsford) [reply articles/2020-lkingsford_writing-your-first-unit-test-with-pytest.md:324]: It can be either. Maybe I'll make it tuples for more clarity in writing.

**Pull request merged**